src/main/java/com/google/code/or/OpenReplicator.java

```diff
@@ -286,6 +289,11 @@ protected void setupHeartbeatPeriod() throws Exception {
 		query.getFirst("SET @master_heartbeat_period = " + nanoSeconds);
 	}
 
+	protected void setupSlaveUUID() throws Exception {
+		final Query query = new Query(this.transport);
+		query.getFirst("SET @slave_uuid = '" + this.uuid + "'");

```

Comment:
is it supposed to be `server_uuid`? Saw that `show slave hosts` mentions a `slave_id` column, but looking at the docs about setting it, just saw references to `server_id`:
http://dev.mysql.com/doc/refman/5.7/en/replication-options.html

Looks like another way would be to set `--replicate-same-server-id` if that is possible (`When starting, the slave I/O thread generates an error and aborts if its master's UUID is equal to its own unless the --replicate-same-server-id option has been set.`)